src/pages/wishlist/wishlist.jsx

```diff
@@ -0,0 +1,95 @@
+import './wishlist.css';
+import { useCart } from '../../context/cart-context';
+import { useAuth } from '../../context/auth-context';
+import axios from "axios"
+
+
+function isItemInCart(cartList, id) {
+    const index = cartList.findIndex((cartListItem) => cartListItem._id === id);
+    if (index !== -1) {
+        return true;
+    }
+    return false;
+}
+
+
+
+export function Wishlist() {
+    const { wishList, cartList, dispatch } = useCart();
+    const { userId } = useAuth();
+
+    async function handleAddToCart(product) {
+        if (isItemInCart(cartList, product._id) === false) {
+            console.log("Item was not in cart")
+            console.log(userId)
+            const response = await axios.post(`https://neog-ecommerce.herokuapp.com/user/${userId}/product/${product._id}`);
+            console.log(response.data);
+            console.log(product)
+            dispatch({
+                type: "ADD_TO_CART",
+                payload: product
+            })
+            dispatch({
+                type: "ADD_TO_CART_FROM_WISHLIST",
+                payload: product
+            })
+        }
+
+    }
+    async function handleRemoveFromWishlist(product) {
+        try {
+            const response = await axios.delete(`https://neog-ecommerce.herokuapp.com/user/${userId}/wishlist/${product._id}`)
+            if (response.status === 202) {
+                console.log(response)
+                const updatedWishlist = response.data.user.wishlist
+                console.log(updatedWishlist)
+                dispatch({ type: "SET_WISHLIST", payload: updatedWishlist })
+
+                // dispatch({ type: "" })
+            }
+            // dispatch({ type: "REMOVE_FROM_CART", payload: product })
+        } catch (error) {
+            console.log(error)
+        }
+        // dispatch({ type: "REMOVE_FROM_WISHLIST", payload: product })
+    }
```

Comment:
Take these two functions in a Utils folder outside.
Hence by now ur Utils folder will have 3 funcs :  `checkIfAlreadyPresent , addToCart, addToWishlist `
This will help in writing DRY code